AppConfigs

AstraLuma opened the pull request:
Closes #53.

Adds support for the use of explicit `AppConfig` references in `INSTALLED_APPS`.

AstraLuma committed:
Allow use of AppConfig

Actually, django has this algorithm in a util. Just use that.

codingjoe commented:
@astronouth7303 thanks for the contribution. It seems like the installation will fail if Django is not installed. Would you mind fixing that?

AstraLuma commented:
... A project called `django-health-check`, only documents how to install on Django, is divided into Django-structured packages, and the majority of tests are Django specific, can be used by other frameworks?

codingjoe commented:
No Django sure is a dependency, that's not the issue. Your patch brakes `pip install` check out the travis output and you'll see what I mean.

AstraLuma committed:
Minimize top-level imports

`__init__.py` is used by `setup.py`, so don't import anything at the top-level.

AstraLuma committed:
pep8

AstraLuma commented:
Ok, taken care of.

AstraLuma committed:
Use an AppConfig to do autodiscovery

codingjoe reviewed (changes requested):
Can you do me a favor and drop the commit that adds the migration, since this is already on master?
Let me know if you need help with that.

AstraLuma commented:
This is why I don't use rebase or squash. It should be the same commit, but for some reason it's not.

AstraLuma commented:
Ok, rebased.

codingjoe commented:
Released in 1.2.0
